gen_cert.py

```diff
@@ -229,7 +217,12 @@ def interstitial_factory():
             self.long_course = long_course or cert_data.get('LONG_COURSE', '').encode('utf-8')
             self.issued_date = issued_date or cert_data.get('ISSUED_DATE', '').encode('utf-8') or 'ROLLING'
             self.interstitial_texts = collections.defaultdict(interstitial_factory())
-            self.interstitial_texts.update(cert_data.get('interstitial', {}))
+            interstitial_dict = {k.encode('utf8'): v.encode('utf8') for k, v in cert_data.get('interstitial', {}).items()}
```

Comment:
I really like list/dict comprehensions, but I find that in-lining them like this make them difficult to read.

Similarly, single-letter variables are opaque and a violation of quality conventions.

```python
interstitial_dict = {
    key.encode('utf8'): value.encode('utf8')
    for key, value in cert_data.get('interstitial', {}).items()
}
```